### DISCUSSION TITLE
How often is node-version updated

### DISCUSSION THREAD

**Discussion by RickyCartner** [coder/code-server#6570] [Q&A] [answered]
I understand that code-server is supposed to update the node-version when vscode does, but what is the general timeline for this? 

There was a security finding on node 18.15.0 and I am being asked to update the version to 18.18.2 (LTS). Can this be done ad-hoc or do we need to wait on a new code-server update (via vscode update). 

If we have to wait, where can I find documentation on the timeline for this? I tried searching through the [vscode issues ](https://github.com/microsoft/vscode/issues) but didn't see anything. 

Any assistance would be appreciated.

**Comment by code-asher** [comment, marked as answer]
Yup, you can replace the Node version ad-hoc, either by replacing the Node binary code-server ships with or by launching code-server with your own Node binary (for example `node /path/to/installed/code-server`, wherever the `package.json` is).

I am not sure what VS Code's process or timeline is for updating.  You can sometimes find information by looking up issues about updates to Electron and then checking what version of Node that version of Electron uses.  We just check the `.yarnrc` on every release.

The main branch [currently targets 18.17.1](https://github.com/microsoft/vscode/blob/main/remote/.yarnrc#L2) which looks like it will be released in 1.86 (next month, I think, because 1.85 was just released).

**Reply by RickyCartner** [reply]
Thank you @code-asher. This was extremely helpful and informative! 

I will look into both of these options and update the thread with my applied results (in case anyone else has this type of issue).
